tests/console/yast2_apparmor.pm

```diff
@@ -38,123 +37,98 @@ sub run {
     }
     assert_screen 'yast2_apparmor_enabled';
     # part 1: open profile mode configuration and check toggle/show all profiles
-    send_key 'alt-n';
+    wait_screen_change { send_key 'alt-n' };
     assert_screen 'yast2_apparmor_profile_mode_configuration';
-    send_key 'alt-o';
-    wait_still_screen(1);
+    wait_screen_change { send_key 'alt-o' };
     assert_screen 'yast2_apparmor_profile_mode_configuration_show_all';
-    wait_still_screen(1);
-    send_key 'tab';
-    wait_still_screen(1);
-    send_key 'down';
-    wait_still_screen(1);
-    send_key 'alt-t';
+    wait_screen_change { send_key 'tab' };
+    wait_screen_change { send_key 'down' };
+    wait_screen_change { send_key 'alt-t' };
     assert_screen 'yast2_apparmor_profile_mode_configuration_toggle';
-    wait_still_screen(1);
-    send_key 'alt-b';
-    wait_still_screen(1);
+    wait_screen_change { send_key 'alt-b' };
 
     # close apparmor configuration
-    send_key 'alt-d';
-    wait_still_screen(1);
+    wait_screen_change { send_key 'alt-d' };
     assert_script_run("systemctl show -p ActiveState apparmor.service | grep ActiveState=active");
 
     # part 2: start apparmor configuration again
-    script_run("yast2 apparmor; echo yast2-apparmor-status-\$? > /dev/$serialdev", 0);
```

Comment:
sorry, but I don't agree with that. `script_run(…, 0)` is equivalent to `type_string "…\n"` so that should not matter. I don't see a point why you would not want to test the exit code. That should be preserved. Please change back to the previous state. If it fails sometimes because it takes longer and you think that the additional waiting time would be acceptable then please just add more to the timeout on the corresponding wait_serial call.